keps/sig-network/2091-cluster-network-policy/README.md

```diff
@@ -0,0 +1,1139 @@
+
+## Summary
+
+Introduce new set of APIs to express an administrator's intent in securing their K8s cluster.
+This doc proposes two new set of resources, ClusterNetworkPolicy API and the DefaultNetworkPolicy API
+to complement the developer focused NetworkPolicy API in Kubernetes.
+
+## Motivation
+
+Kubernetes provides NetworkPolicy resources to control traffic within a cluster.
+NetworkPolicies focus on expressing a developers intent to secure their applications.
+Thus, in order to satisfy the needs of a security admin, we propose to introduce new set of APIs
+that capture the administrators intent.
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
+
+The goals for this KEP are to satisfy two key user stories:
+1. As a cluster administrator, I want to enforce irrevocable guardrails that all workloads must adhere to in order to guarantee the safety of my clusters.
+2. As a cluster administrator, I want to deploy a default set of policies to all workloads that may be overridden by the developers if needed.
+
+There are several unique properties that we need to add in order accomplish the user stories above.
+1. Deny rules and, therefore, hierarchical enforcement of policy
+2. Semantics for a cluster-scoped policy object that may include namespaces/workloads that have not been created yet.
+3. Backwards compatibility with existing Kubernetes Network Policy API
+
+### Non-Goals
+
+Our mission is to solve the most common use cases that cluster admins have. That is, we don't want to solve for every possible policy permutation a user can think of. Instead, we want to design an API that addresses 90-95% use cases while keeping the mental model easy to understand and use.
+
+Additionally, this proposal is squarely focused on solving the needs of the Cluster Administrator. It is not intended to solve:
+1. Logging / error reporting for Network Policy
+2. Kubernetes Node policies
+3. New policy selectors (services, service accounts, etc.)
+
+## Proposal
+
+In order to achieve the two primary broad use cases for a cluster admin to secure K8s clusters,
+we propose to introduce the following two new resources under `networking.k8s.io` API group:
+- ClusterNetworkPolicy resource
+- DefaultNetworkPolicy resource
+
+### ClusterNetworkPolicy resource
+
+A ClusterNetworkPolicy resource will help the administrators set strict security rules for the cluster,
+i.e. a developer CANNOT override these rules by creating NetworkPolicies that applies to the same workloads
+as the ClusterNetworkPolicy does.
+
+Unlike the NetworkPolicy resource in which each rule represents a whitelisted traffic, ClusterNetworkPolicy
```

Comment:
would change the term whitelisted to allow listed here (or maybe some other term)

Reply:
thanks.. done